src/net/net.c

```diff
@@ -95,6 +95,48 @@ int net_dst_source_addr_get(struct sa *dst, struct sa *ip)
 }
 
 
+/**
+ * Check if connection from source to destination is possible
+ *
+ * @param src Source IP address
+ * @param dst Destination IP address
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int net_check_conn(const struct sa *src, const struct sa *dst)
+{
+	struct udp_sock *us;
+	struct sa srctmp;
+	struct sa dsttmp;
+	int port;
+	int err = 0;
+
+	if (!sa_isset(src, SA_ADDR) || !sa_isset(dst, SA_ADDR))
+		return EINVAL;
+
+	srctmp = *src;
+	dsttmp = *dst;
+	sa_cpy_scopeid(&dsttmp, src);
+	for (port = sa_port(src) + 3; port < sa_port(src) + 23; port++) {
+		err = udp_listen(&us, &srctmp, NULL, NULL);
+		if (!err)
+			break;
+
+		sa_set_port(&srctmp, port);
```

Comment:
Shouldn`t `sa_set_port` before udp_listen? Or is listening on src original port in every case harmless?

Reply:
Is the loop needed? Maybe listening on port 0 to get a free port from system is enough?